google-cloud-bigquery/acceptance/bigquery/dataset_access_test.rb

```diff
@@ -161,4 +162,292 @@
       refute dataset.access.reader_routine? routine
     end
   end
+
+  describe :condition do
+    let(:dataset_id) { "#{SecureRandom.hex(4)}_dataset" }
+    let(:dataset) do
+      d = bigquery.dataset dataset_id, access_policy_version: 3
+      if d.nil?
+        d = bigquery.create_dataset dataset_id, access_policy_version: 3
+      end
+      d
+    end
+
+    let(:condition) do
+      Google::Cloud::Bigquery::Condition.new(
+        "true",
+        description: "example description",
+        location: "path/to/example/location",
+        title: "example title"
+      )
+    end
+    let(:c) { condition.to_gapi.to_h }
+    let(:group_val) { "ruby-cloud-eng@google.com" }
+    let(:iam_member_val) { "group:ruby-cloud-eng@google.com" }
+    let(:domain_val) { "google.com" }
+    
+    it "adds a reader access entry with specifying user scope and condition" do
+      dataset.access do |acl|
+        acl.add_reader_user user_val, condition: condition
+      end
+
+      dataset = bigquery.dataset dataset_id, access_policy_version: 3
```

Comment:
Here you are replacing the `dataset` you used a few lines above. (The previous `dataset` uses the one created from `let(:dataset)`.) Is that your intent? It's a little confusing to do this and have a local variable like this shadow a previously used name. If your intent is to test that you can re-acquire the resource and assert it has the correct state, consider using a different name like `reread_dataset` to make it clear that's what you're doing.